ProjectLighthouse.Servers.API/Controllers/UserEndpoints.cs

```diff
@@ -50,6 +50,32 @@ public async Task<IActionResult> GetUser(string username)
         return this.Ok(user);
     }
 
+    /// <summary>
+    /// Searches for the user based on the query
+    /// </summary>
+    /// <param name="query">The search query</param>
+    /// <returns>A list of users</returns>
+    /// <response code="200">The list of users, if any were found</response>
+    /// <response code="404">No users matched the query</response>
+    [HttpGet("user_search")]
```

Comment:
Shouldn't this be something like `/user/search`?

Reply:
maybe, though I wasn't sure if that would clash with the endpoint that's already bound to /user/{id}

Reply:
maybe it should be under a /search endpoint then, like `/api/v1/search/user?query=`?